public/javascripts/PrairieEditor.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * PrairieEditor is currently a simple jQuery script that sets up ace editors
+ * and configures a save button for each file. It also previews the question by
+ * asynchronously rendering variants and grading submissions.
+ */
+$(function() {
+
+    // Ensure the CSRF token has been set
+    if (! document.csrfToken)
+        console.error('No CSRF token set, asynchronous requests will fail.');
+
+    // Creates an XMLHttpRequest object that posts to the same URL as the editor
+    var post = function(data, onComplete) {
+        var xhr = new XMLHttpRequest();
+        xhr.open("POST", document.URL, true);
+        xhr.setRequestHeader('x-csrf-token', document.csrfToken);
+        xhr.setRequestHeader("Content-Type", "application/json");
+
+        xhr.onreadystatechange = function() {
+            if (xhr.readyState == 4 && xhr.status == 200 && onComplete != null)
+                onComplete(xhr.responseText);
+        };
+
+        if (data)
+            xhr.send(JSON.stringify(data));
+
+        return xhr;
+    };
+
+    // For each editable zone
+    $('[data-edit]').each(function() {
+        // Get file data
+        var fileName = $(this).data('edit');
+        var fileMode = $(this).data('mode');
+
+        // Get IDs of editor and save button
+        var editorId = $(this).attr('id');
+        var saveId = $(this).data('save');
+        var saveButton = $('#' + saveId);
+
+        // Create an Ace editor instance
+        var editor = ace.edit(editorId);
+        editor.setTheme('ace/theme/monokai');
```

Comment:
We've generally use light-themed components everywhere in PL, so we should think about what we want here.